### PR TITLE
Add support for CSE Custom Entity Type and Entity Criticality Config

### PR DESCRIPTION
**Pull request by pmontiel-sumo**
Add support for CSE Custom Entity Type and Entity Criticality Config

### CODE REVIEW

File: website/docs/r/cse_custom_entity_type.html.markdown [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  fields =["file_hash_md5", "file_hash_sha1"]
+}
+```
+
+## Argument reference
+
+The following arguments are supported:
+
+- `name` - (Required) Human friend and unique name. Examples: "Ip Address", "Username", "Mac Address".
+- `identifier` - (Required) Machine friendly and unique identifier. Examples: "ip", "username", "mac".
```
Comment (josh-williams) [inline website/docs/r/cse_custom_entity_type.html.markdown:25]: "username","ip","mac", won't actually be valid here. Let's go with the example of `name="File Hash"`, `identifier="filehash"`

Comment (pmontiel-sumo) [reply website/docs/r/cse_custom_entity_type.html.markdown]: done under:
https://github.com/SumoLogic/terraform-provider-sumologic/pull/275/commits/0d141b9253b64f0aa25854a15d86d0f1ddfb1fbb

File: sumologic/resource_sumologic_cse_custom_entity_type.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		Update: resourceSumologicCSECustomEntityTypeUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
```
Comment (seansain) [inline sumologic/resource_sumologic_cse_custom_entity_type.go:22]: No need for `ForceNew` if it's false

File: sumologic/resource_sumologic_cse_entity_criticality_config.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		Update: resourceSumologicCSEEntityCriticalityConfigUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
```
Comment (seansain) [inline sumologic/resource_sumologic_cse_entity_criticality_config.go:22]: same here with `ForceNew`

File: sumologic/resource_sumologic_cse_entity_criticality_config.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+		})
+
+		if err != nil {
+			return err
+		}
+
+		d.SetId(id)
+	}
+
+	return resourceSumologicCSEEntityCriticalityConfigUpdate(d, meta)
```
Comment (seansain) [inline sumologic/resource_sumologic_cse_entity_criticality_config.go:80]: `return resourceSumologicCSEEntityCriticalityConfigRead(d, meta)`

File: sumologic/resource_sumologic_cse_entity_criticality_config_test.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+				Config: testCreateCSEEntityCriticalityConfigConfig(nName, nSeverityExpression),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckCSEEntityCriticalityConfigExists(resourceName, &entityCriticalityConfig),
+					testCheckEntityCriticalityConfigValues(&entityCriticalityConfig, nName, nSeverityExpression),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+				),
+			},
+		},
+	})
+}
```
Comment (seansain) [inline sumologic/resource_sumologic_cse_entity_criticality_config_test.go:31]: we should add tests for importing and updating?

File: sumologic/resource_sumologic_cse_custom_entity_type.go [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+		})
+
+		if err != nil {
+			return err
+		}
+
+		d.SetId(id)
+	}
+
+	return resourceSumologicCSECustomEntityTypeUpdate(d, meta)
```
Comment (seansain) [inline sumologic/resource_sumologic_cse_custom_entity_type.go:90]: `return resourceSumologicCSECustomEntityTypeRead(d, meta)`

File: sumologic/resource_sumologic_cse_custom_entity_type_test.go [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+				Config: testCreateCSECustomEntityTypeConfig(nName, nIdentifier, nFields),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckCSECustomEntityTypeExists(resourceName, &customEntityType),
+					testCheckCustomEntityTypeValues(&customEntityType, nName, nIdentifier, nFields),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+				),
+			},
+		},
+	})
+}
```
Comment (seansain) [inline sumologic/resource_sumologic_cse_custom_entity_type_test.go:32]: add tests for import and update?

File: sumologic/resource_sumologic_cse_custom_entity_type.go (hunk 2 above)
Comment (pmontiel-sumo) [reply sumologic/resource_sumologic_cse_custom_entity_type.go]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/275/commits/6f2a00385fe4868446463043e99183b6d4b80bd7

File: sumologic/resource_sumologic_cse_custom_entity_type.go (hunk 6 above)
Comment (pmontiel-sumo) [reply sumologic/resource_sumologic_cse_custom_entity_type.go]: done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/275/commits/6f2a00385fe4868446463043e99183b6d4b80bd7

File: sumologic/resource_sumologic_cse_custom_entity_type_test.go (hunk 7 above)
Comment (pmontiel-sumo) [reply sumologic/resource_sumologic_cse_custom_entity_type_test.go:32]: added update test under
https://github.com/SumoLogic/terraform-provider-sumologic/pull/275/commits/eaf35c35121c9da0e0db4f53f0ffe2d8c937c064

File: sumologic/resource_sumologic_cse_entity_criticality_config_test.go (hunk 5 above)
Comment (pmontiel-sumo) [reply sumologic/resource_sumologic_cse_entity_criticality_config_test.go:31]: added update test under
https://github.com/SumoLogic/terraform-provider-sumologic/pull/275/commits/eaf35c35121c9da0e0db4f53f0ffe2d8c937c064

**Review by josh-williams — APPROVED** [review]

**Review by seansain — APPROVED** [review]
lgtm

**Pull request merged**